### DISCUSSION TITLE
내보내기를 한 후 내보내기 당한 사람 메세지를 특정 명령어로 전부 한번에 가리기 하는 방법이 있을까요?

### DISCUSSION THREAD

**Discussion by gangdaegu** [storycraft/node-kakao#799] [Q&A] [answered]
a
a
a
a
a
.
.
.
a

이렇게 도배한 사람을 내보내기를 통해 내보내고
남은 메세지를 수동으로 지금까지 지우기를 하고 있었는데..

특정 명령어를 통해 하나하나씩이 아닌 한꺼번에 가리기 하는 방법이 있을까요?

아니면 그 사람을 일단 채팅잠금 처리 해놓고 나서 지금까지 보낸 메세지를 다 가리는 방법이라던가..

혹시 있을까요?

**Comment by dbswlstjr2** [comment, marked as answer]
가능합니다 
대신 당신이 직접 소스를 짜야죠

**Reply by gangdaegu** [reply]
코딩은 할 수 있는데 어떤 방식으로 해야할지 몰라서 감을 좀 잡고자 질문드렸는데
혹시 어떤 방식으로 하면 좋을까요?

**Reply by dbswlstjr2** [reply]
그냥 당신의 명령어를 이용하면 답장형식 또는 멘션 형식으로 강퇴를 하시지 않나요? 그걸이용해서 답장된사람 혹은 멘션된사람이 /강퇴 명령어를 당했으면 그 사람의 유저아이디를 뽑을수있으니 그사람이 쓴 글만 지울수있게 가리기를 반복하셔요

**Comment by gogop45** [comment]
채팅금지 (채금)

**Comment by benesiaxd** [comment]
당연히 방법이있습니다.

**Reply by gangdaegu** [reply]
저번주에 할 수 있다는 말 듣고 기능 추가하였습니다ㅎㅎ

**Comment by gogop45** [comment]
그런거 저도 써보고싶네요... 코드는 어디서 받아볼수있을까요?

**Reply by dbswlstjr2** [reply]
직접 코딩해야죠 모듈을 찾아보면서 ㅎㅅㅎ

**Reply by gangdaegu** [reply]
문서 찾아보고 응용 하시면 가능합니다!

**Reply by gogop45** [reply]
도와주시는분 1도없네

**Reply by gangdaegu** [reply]
> 도와주시는분 1도없네

소스를 너무 대놓고 보여드릴 수는 없고 공부를 위해서라면 어찌 제가 만들었는지는 알려드릴 수 있습니다.
저 같은 경우 채팅방에서 메세지를 보내면 유저별로 메세지를 array에다가 담아두고 강퇴를 했을 시 강퇴한 유저 아이디값을 뽑아서 그걸로 아까 만들어놓은 유저별 array 에서 해당 유저를 찾은 다음에 hideChat으로 반복 돌면서 처리해놓았습니다.

**Reply by benesiaxd** [reply]
> 도와주시는분 1도없네

남들이 공부하고 시간써서 짠걸 그냥달라하는게 정상인가요
하루전에 질문하셨는데 하루동안 고민하고 찾아보셨으면 지금쯤 더한것도 짜는게 가능했을텐데요

**Reply by gogop45** [reply]
kickUser> removeKicked 에관련넣었는데 안되서 말입니다

**Reply by artist012** [reply]
> kickUser> removeKicked 에관련넣었는데 안되서 말입니다

?
